dashboard/app/controllers/sessions_controller.rb

```diff
@@ -73,8 +73,8 @@ def lockout
       @disallowed_email = current_user.hashed_email
     end
 
-    # Determine the deletion date as the creation time of the account + 7 days
-    @delete_date = current_user.created_at.since(7.days)
+    # Determine the deletion date as the lockout date of the account + 7 days
+    @delete_date = Date.parse(current_user.child_account_compliance_lock_out_date).since(7.days)
```

Comment:
Is it possible for `child_account_compliance_lock_out_date` to be nil? `Date.parse` will raise an error if so

Reply:
They should have a lockout date if they are rendering the lockout page. It is set when their lockout state is set, which is what the ComplianceState.locked_out? is looking for.

Reply:
They should be redirected away if they aren't actually locked out, which should always be accompanitd by having a `child_account_compliance_lock_out_date`. I think it might be possible to get it to be nil if their lockout status was set manually instead of using the lock_out method? But in the wild that shouldn't happen.

Reply:
The only problem is that `Date.parse` drops the time and timezone information and I have to think about whether or not this is a problem.

Reply:
You may want `DateTime.parse` but I'm not entirely sure... but that aligns more with the original `created_at.since` which would give you a `DateTime` and not just a `Date`

```
[development] dashboard > Date.parse(u.child_account_compliance_lock_out_date).iso8601
=> "2024-06-30"
[development] dashboard > DateTime.parse(u.child_account_compliance_lock_out_date).iso8601
=> "2024-06-30T16:52:58-07:00"
```

Reply:
We could also do a DateTime, but I figured we only need date level precision, and it's an approximation for display purposes, not the actual mechanism for deciding when to delete

Reply:
Yep, just requires me to lose a few braincells to think about timezone implications, since it technically introduces them.

I think it is ok because the timezone as recorded (in mine it is -7) is the same that they want to actually see. Tested with a timestamp that would be the next day in UTC and it parsed as expected to local time just fine.

So, all good!

Reply:
@wilkie it looks like the panel is [setting it to UTC](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/sessions/LockoutPanel.tsx#L250). The lockout date appears to be set based on server time, which is likely US eastern time. Should we make any changes here, or would you rather I just change it to use a DateTime, and preserve the previous logic as much as possible?

Reply:
It's probably safest and most dogmatic to just push the most accurate information (`DateTime`) out and decide on its actual representation on the front-end.

Looks like you need to set the user property in the controller test. easy peasy.